source/LocationFilter.cpp

```diff
@@ -97,10 +104,20 @@ void LocationFilter::Load(const DataNode &node)
 		else if(child.Token(0) == "government")
 		{
 			for(int i = 1; i < child.Size(); ++i)
-				governments.insert(GameData::Governments().Get(child.Token(i)));
+			{
+				if(child.Token(i).c_str()[0] == '!')
```

Comment:
I wonder if you should use PrintTrace to warn the mission writer if they have supplied both a include and exclude government filter. As mentioned previously, it doesn't make sense to use both types in a single mission, since a system can have only one government.
Furthermore, a single line multiple exclude probably shouldn't be allowed, e.g. `!Syndicate !Republic` means any system matches.